cmd/mt-index-cat/main.go

```diff
@@ -237,53 +283,92 @@ func main() {
 		}
 	}
 
-	var defs []schema.MetricDefinition
-	if len(partitions) == 0 {
-		defs = idx.Load(nil, time.Now())
-	} else {
-		defs = idx.LoadPartitions(partitions, nil, time.Now())
-	}
-	// set this after doing the query, to assure age can't possibly be negative unless if clocks are misconfigured.
-	out.QueryTime = time.Now().Unix()
-	total := len(defs)
-	shown := 0
-
-	for _, d := range defs {
-		// note that prefix and substr can be "", meaning filter disabled.
-		// the conditions handle this fine as well.
-		if !strings.HasPrefix(d.Name, prefix) {
-			continue
-		}
-		if !strings.HasSuffix(d.Name, suffix) {
-			continue
-		}
-		if !strings.Contains(d.Name, substr) {
-			continue
-		}
-		if tags == "none" && len(d.Tags) != 0 {
-			continue
-		}
-		if tags == "some" && len(d.Tags) == 0 {
-			continue
+	// if partitionStr is set to all (*) and we are using bigtable then we must
+	// ensure that we know the total number of partitions
+	if partitionStr == "*" && btI > 0 {
+		if btTotalPartitions == -1 {
+			log.Println("When selecting all partitions with bigtable you must specify the total number of partitions for the instance")
+			flag.Usage()
+			os.Exit(-1)
+		} else {
+			for i := 0; i < btTotalPartitions; i++ {
+				partitions = append(partitions, int32(i))
+			}
 		}
-		if regex != nil && !regex.MatchString(d.Name) {
-			continue
+	}
+
+	var total int
+	var shown int
+
+	processDefs := func(defs []schema.MetricDefinition) {
+		total += len(defs)
+		if shown >= limit && limit > 0 {
+			log.Infof("Limit (%d) reached while processing Metric Definitions", limit)
```

Comment:
note that by printing to stdout here, and below, we break the output format.
a common use case is to pipe the output of mt-index-cat into other tool, so i'm hesitant of adding these new log messages. (note that all other/pre-existing log calls are error messages or help output

Reply:
also this check here is redundant since you already check every time after showing and we only call processDefs once.

Reply:
The check is not redundant. While processing multiple BT partitions it is called once per partition. This is why the check is also at the beginning. The other way to do it would be to place that logic / check where the BT partitions are processed, but it really doesn't matter.

I will look into changing the loglines